src/mesh/parallel/fci.cxx

```diff
@@ -169,7 +169,7 @@ FCIMap::FCIMap(Mesh& mesh, Options& options, int offset_, BoundaryRegionPar* bou
 
         // The integer part of xt_prime, zt_prime are the indices of the cell
         // containing the field line end-point
-        i_corner(x, y, z) = static_cast<int>(floor(xt_prime(x, y, z)));
+        i_corner(x, y, z) = static_cast<int>(floor(std::abs(xt_prime(x, y, z))));
```

Comment:
I think it is the wrong result, as a negative xt_prime is very much allowed and the only way to get into the inner x boundaries. It seems like `git blame` points to @d7919 for this one.  (I might be wrong).

Reply:
Yes it looks like I introduced this in commit 122c04be5e5ce9881c47a6136866906164d732dd. Unfortunately my commit message wasn't super clear but it suggests there was a change in Zoidberg around that time that triggered me to make this change.

Reply:
From what I can tell, it looks like the idea was to indicate boundaries by changing the sign of `xt_prime`, rather than just setting it to exactly -1 (04faac4). This doesn't seem to have survived though -- I'm not sure if it's been clobbered by a merge or what.

It sounds like the intention was to allow interpolation in boundaries, given the message in 122c04b. The same commit that adds `abs` also changes `boundary_mask(x, y, z) = false;` instead of `true` (L185 in 122c04b). This causes the interpolation to not skip any points, when it previously skipped boundaries (as these would be filled in by the parallel boundary conditions).

I'm not sure the reasoning behind that, and if it's required for 3D metrics? I'm tempted to revert those changes, at least for now.